src/vim.js

```diff
@@ -1123,6 +1147,82 @@ export function initVim(CodeMirror) {
       }
     }
 
+    // langmap support
+    function updateLangmap(langmapString) {
+      if (langmap != null && langmap.string == langmapString) return;
+      langmap = parseLangmap(langmapString);
+    }
+    function langmapIsLiteralMode(vim) {
+      // Determine if keystrokes should be interpreted literally
+      return vim.insertMode;
+    }
+    function parseLangmap(langmapString) {
```

Comment:
Could you please use the shorter parsing code that i have provided in the issue?

Reply:
Will do